File: My_menu_java/app/src/main/java/com/example/my_menu_java/aty1.java
package com.example.my_menu_java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class aty1 extends AppCompatActivity {
    Button btn;
    TextView t_drink;
    TextView t_sugar;
    TextView t_ice;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main1);
        btn = (Button) findViewById(R.id.btn_choice);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent();
                i.setClass(aty1.this,aty2.class);
                startActivityForResult(i,0);
            }
        });
    }
    protected void onActivityResult(int requestCode,int resultCode,Intent data) {
        if(requestCode == 0){
            if(resultCode == 101){
                Bundle b = data.getExtras();
                String str1 = b.getString("drink_level");
                String str2 = b.getString("sugar_level");
                CharSequence str3 = b.getString("ice_level");

                t_drink = findViewById(R.id.tv_drink_null);
                t_sugar = findViewById(R.id.tv_sugar_null);
                t_ice = findViewById(R.id.tv_ice_null);

                t_drink.setText(str1);
                t_sugar.setText(str2);
                t_ice.setText(str3);
            }
        }
    }
}

File: My_menu_kotlin/app/src/main/java/com/example/my_menu_kotlin/aty1.kt
package com.example.my_menu_kotlin

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.main1.*

class aty1 : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main1)
        val intent = Intent(this,aty2::class.java)
        btn_choice.setOnClickListener {
            startActivityForResult(intent,0)
        }
    }
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        val b = data?.extras?:return
        if(requestCode == 0){
            if(resultCode == 101){
                val str1 = b.getString("drink_level")
                val str2 = b.getString("sugar_level")
                val str3 = b.getString("ice_level")

                val t_drink = findViewById<TextView>(R.id.tv_drink_null)
                val t_sugar = findViewById<TextView>(R.id.tv_sugar_null)
                val t_ice = findViewById<TextView>(R.id.tv_ice_null)

                t_drink.text=str1
                t_sugar.text=str2
                t_ice.text=str3

            }
        }
    }
}
